pytorch_lightning/trainer/progress.py

```diff
@@ -11,12 +11,31 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from dataclasses import dataclass, field
+from dataclasses import asdict, dataclass, field
 from typing import Optional
 
 
 @dataclass
-class Tracker:
+class _DataclassStateDictMixin:
+
+    def __getstate__(self) -> dict:
+        return asdict(self)
+
+    def __setstate__(self, state: dict) -> None:
+        self.__dict__.update(state)
+
+    def state_dict(self) -> dict:
+        return self.__getstate__()
+
+    @classmethod
+    def load_state_dict(cls, state_dict: dict) -> "_DataclassStateDictMixin":
```

Comment:
I think this is not consistent with the rest of the load_state_dict semantics where we operate on an instance.
I would like to suggest simply renaming this to `from_state_dict`.
Optionally a separate load_state_dict method on an instance could also be useful.

Reply:
Good point. Will change it